testing/fil_token_integration/actors/test_actor/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+            }
+
+            // get token transfer data
+            let token_params: FRC46TokenReceived = params.payload.deserialize().unwrap();
+
+            // todo: examine the operator_data to determine our next move
+            let action: TestAction = token_params.operator_data.deserialize().unwrap();
+            match action {
+                TestAction::Accept => {
+                    // do nothing, return success
+                    NO_DATA_BLOCK_ID
+                }
+                TestAction::Reject => {
+                    // abort to reject transfer
+                    sdk::vm::abort(
+                        ExitCode::USR_FORBIDDEN.value(),
+                        Some("rejecting transfer"),
+                    );
+                }
+                TestAction::Transfer(to, operator_data) => {
+                    // transfer to a target address
+                    let transfer_params = TransferParams {
+                        to,
+                        amount: token_params.amount,
+                        operator_data,
+                    };
+                    let receipt = sdk::send::send(&Address::new_id(sdk::message::caller()), method_hash!("Transfer"), RawBytes::serialize(&transfer_params).unwrap(), TokenAmount::zero()).unwrap();
+                    // ignore failures at this level and return the transfer call receipt so caller can decide what to do
+                    return_ipld(&receipt)
+                }
+                TestAction::Burn => {
+                    // burn the tokens
+                    let burn_params = BurnParams {
+                        amount: token_params.amount,
+                    };
+                    let receipt = sdk::send::send(&Address::new_id(sdk::message::caller()), method_hash!("Burn"), RawBytes::serialize(&burn_params).unwrap(), TokenAmount::zero()).unwrap();
+                    if !receipt.exit_code.is_success() {
+                        panic!("burn call failed");
+                    }
+                    NO_DATA_BLOCK_ID
+                }
+            }
+        },
+        "Action" => {
+            // take action independent of the receiver hook
+            let params: ActionParams = deserialize_params(input);
+
+            // get our balance
+            let get_balance = || {
+                let self_address = Address::new_id(sdk::message::receiver());
+                let balance_receipt = sdk::send::send(&params.token_address, method_hash!("BalanceOf"), RawBytes::serialize(self_address).unwrap(), TokenAmount::zero()).unwrap();
+                if !balance_receipt.exit_code.is_success() {
+                    panic!("unable to get balance");
+                }
+                balance_receipt.return_data.deserialize::<TokenAmount>().unwrap()
+            };
+
+            match params.action {
+                TestAction::Accept => {
+                    // nothing to do here
+                    NO_DATA_BLOCK_ID
```

Comment:
Abort for safety? This means the caller did somethign wrong

Reply:
Will do